### DISCUSSION TITLE
Visual Studio 2019/2022 Trouble: How to use v142 toolset specifically when both VS2022 and VS2019 are installed

### DISCUSSION THREAD

**Discussion by imikejackson** [microsoft/vcpkg#30236] [Q&A]
I have been struggling to get this to work. We have a GitHub CI where we are building against the v142 and v143 toolsets. The issue is that while there is claimed compatibility between the 2 toolsets (for linking and runtime) there isn't. When we try the v142 toolset, VCPKG insists on using the v143 toolsets to build our dependencies. ITK (https://github.com/InsightSoftwareConsortium/ITK) is the main culprit at this point. When the CI gets to our compile step the compile of our code will use the v142 tool set. When the final linking is attempting I get the following error:

```
     Creating library D:/a/complex/complex/build/bin/Release/ITKImageProcessing.lib and object D:/a/complex/complex/build/bin/Release/ITKImageProcessing.exp
itksys-5.2.lib(SystemTools.obj) : error LNK2019: unresolved external symbol __std_find_trivial_2 referenced in function "wchar_t * __cdecl __std_find_trivial<wchar_t,wchar_t>(wchar_t *,wchar_t *,wchar_t)" (??$__std_find_trivial@_W_W@@YAPEA_WPEA_W0_W@Z) [D:\a\complex\complex\build\Plugins\ITKImageProcessing\ITKImageProcessing.vcxproj]
D:\a\complex\complex\build\bin\Release\ITKImageProcessing.complex : fatal error LNK1120: 1 unresolved externals [D:\a\complex\complex\build\Plugins\ITKImageProcessing\ITKImageProcessing.vcxproj]
```

That comes from this run: https://github.com/BlueQuartzSoftware/complex/actions/runs/4432239443/jobs/7776058286#step:8:2293

We have a system in the office with only VS 2019 installed and the codes will compile and link correctly and without error.

I have tried a number of items in the CMakePresets.json file to communicate to VCPKG to use the VS2019 v142 tool set but nothing seems to work.

The specific PR causing this issue is at: https://github.com/BlueQuartzSoftware/complex/pull/482

Thanks to anyone that could provide guidance.

**Comment by autoantwort** [comment]
You have to create a custom triplet and set https://learn.microsoft.com/en-us/vcpkg/users/triplets#vcpkg_platform_toolset

**Reply by HppZ** [reply]
> You have to create a custom triplet and set https://learn.microsoft.com/en-us/vcpkg/users/triplets#vcpkg_platform_toolset

shit

**Comment by Moin2002-tech** [comment]


**Reply by Osyotr** [reply]
AI slop. `VCPKG_PLATFORM_TOOLSET` only makes sense in triplet.
